libiop/bcs/hashing/hash_enum.tcc

```diff
@@ -164,4 +163,53 @@ two_to_one_hash_function<hash_type> get_two_to_one_hash(const bcs_hash_type hash
     return get_two_to_one_hash_internal<hash_type, FieldT>(FieldT::zero(), hash_enum, security_parameter);
 }
 
+/* Hash digest 2^n->1 hash. */
+template<typename hash_type, typename FieldT>
+cap_hash_function<hash_type> get_cap_hash_internal(
+    const typename libff::enable_if<!std::is_same<hash_type, FieldT>::value, FieldT>::type _,
+    const bcs_hash_type hash_enum,
+    const size_t security_parameter)
+{
+    if (hash_enum == blake2b_type)
+    {
+        return blake2b_many_to_one_hash;
+    }
+    throw std::invalid_argument("bcs_hash_type unknown (blake2b cap hash)");
 }
+
+/* Algebraic 2^n->1 hash. */
+template<typename hash_type, typename FieldT>
+cap_hash_function<FieldT> get_cap_hash_internal(
+    const typename libff::enable_if<std::is_same<hash_type, FieldT>::value, FieldT>::type _,
+    const bcs_hash_type hash_enum,
+    const size_t security_parameter)
+{
+    if (hash_enum == starkware_poseidon_type || hash_enum == high_alpha_poseidon_type)
+    {
+        if (security_parameter != 128)
+        {
+            throw std::invalid_argument("Poseidon only supported for 128 bit soundness.");
+        }
+        poseidon_params<FieldT> params = get_poseidon_parameters<FieldT>(hash_enum);
+        /* security parameter is -1 b/c */
```

Comment:
Clarify "-1" and "b/c"?

Reply:
I have no idea what it means, it was copied from elsewhere

Reply:
Then this line should be removed.